lang/cs/Org.Apache.REEF.IMRU/API/IMRUJobDefinitionBuilder.cs

```diff
@@ -241,6 +258,16 @@ public IMRUJobDefinition Build()
                 Exceptions.Throw(new NullReferenceException("Update function configuration cannot be null"), Logger);
             }
 
+            if (null == _perMapConfigGeneratorConfig || _perMapConfigGeneratorConfig.Count == 0)
```

Comment:
Same as above: You can probably remove this code. Also, if you assign `_perMapConfigGeneratorConfig` at declaration or in the constructor, you can make it `readonly` to make sure it never becomes `null`.